app/src/main/java/com/babylon/wallet/android/data/dapp/IncomingRequestRepository.kt

```diff
@@ -68,25 +72,71 @@ class IncomingRequestRepositoryImpl @Inject constructor() : IncomingRequestRepos
         }
     }
 
+    /**
+     * There are two path of execution when using this method:
+     * - there is high priority screen in the queue, so the incoming request is added below it,
+     * taking priority over requests currently in the queue
+     * - there is no high priority screen: request is added at the top of queue and if there other request currently handled,
+     * we send a defer event for it so that UI can react and defer handling, without removing it from the queue.
+     * Deferred request will be handled again when top priority one handling completes
+     */
+    override suspend fun addMobileConnectRequest(incomingRequest: IncomingRequest) {
```

Comment:
Maybe `addDeferredRequest` instead of naming it to mobile connect

Reply:
this is not deferred request, mobile connect ones take priority over others